kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/clusternetworkaddressconfigprovider/PortPerBrokerClusterNetworkAddressConfigProvider.java

```diff
@@ -104,17 +105,20 @@ public static class PortPerBrokerClusterNetworkAddressConfigProviderConfig {
         private final HostPort bootstrapAddress;
         private final String brokerAddressPattern;
         private final int brokerStartPort;
+        private final int firstNodeId;
         private final int numberOfBrokerPorts;
 
         public PortPerBrokerClusterNetworkAddressConfigProviderConfig(@JsonProperty(required = true) HostPort bootstrapAddress,
                                                                       @JsonProperty(required = false) String brokerAddressPattern,
                                                                       @JsonProperty(required = false) Integer brokerStartPort,
+                                                                      @JsonProperty(required = false, defaultValue = "0") Integer firstBrokerId,
                                                                       @JsonProperty(required = false, defaultValue = "3") Integer numberOfBrokerPorts) {
             Objects.requireNonNull(bootstrapAddress, "bootstrapAddress cannot be null");
 
             this.bootstrapAddress = bootstrapAddress;
             this.brokerAddressPattern = brokerAddressPattern != null ? brokerAddressPattern : bootstrapAddress.host();
             this.brokerStartPort = brokerStartPort != null ? brokerStartPort : (bootstrapAddress.port() + 1);
+            this.firstNodeId = firstBrokerId != null ? firstBrokerId : 0;
```

Comment:
I've got mixed naming here, but it's something to discuss. Currently the proxy is aware of brokers, but potentially in future we might need to proxy controller only nodes. Maybe we should move to the newer node and nodeId language used by Kafka?

Reply:
I think you are right.  Adopting the node language will be right thing now controllers are exposed to clients too.  But this can wait, we need to revisit port-per-broker, so I think we can sweep up the naming issues at the same time.